src/atomate2/aims/jobs/base.py

```diff
@@ -0,0 +1,313 @@
+        """
+        # the structure transformation part was deleted; can be reinserted when needed
+        if isinstance(structure, (Structure, Molecule)):
+            atoms = MSONableAtoms.from_pymatgen(structure)
+        else:
+            atoms = structure.copy()
+
+        # copy previous inputs if needed (governed by self.copy_aims_kwargs)
+        if prev_dir is not None:
+            copy_aims_outputs(prev_dir, **self.copy_aims_kwargs)
+
+        # write aims input files
+        self.write_input_set_kwargs["prev_dir"] = prev_dir
+        write_aims_input_set(
+            atoms, self.input_set_generator, **self.write_input_set_kwargs
+        )
+
+        # write any additional data
+        for filename, data in self.write_additional_data.items():
+            dumpfn(data, filename.replace(":", "."))
+
+        # run FHI-aims
+        run_aims(**self.run_aims_kwargs)
+
+        # parse FHI-aims outputs
+        task_doc = AimsTaskDoc.from_directory(Path.cwd(), **self.task_document_kwargs)
+        task_doc.task_label = self.name
+
+        # decide whether child jobs should proceed
+        stop_children = should_stop_children(task_doc, **self.stop_children_kwargs)
+
+        # cleanup files to save disk space
+        cleanup_aims_outputs(directory=Path.cwd())
+
+        return Response(
+            stop_children=stop_children,
+            output=task_doc if self.store_output_data else None,
+        )
+
+
+@dataclass
+class ConvergenceMaker(Maker):
+    """Defines a convergence workflow with a maximum number of steps.
+
+    A job that performs convergence run for a given number of steps. Stops either
+    when all steps are done, or when the convergence criterion is reached, that is when
+    the absolute difference between the subsequent values of the convergence field is
+    less than a given epsilon.
+
+    Parameters
+    ----------
+    name : str
+        A name for the job
+    maker: .BaseAimsMaker
+        A maker for the run
+    criterion_name: str
+        A name for the convergence criterion. Must be in the run results
+    epsilon: float
+        A difference in criterion value for subsequent runs
+    convergence_field: str
+        An input parameter that changes to achieve convergence
+    convergence_steps: Iterable
+        An iterable of the possible values for the convergence field.
+        If the iterable is depleted and the convergence is not reached,
+        that the job is failed
+    """
+
+    name: str = "Convergence job"
+    maker: BaseAimsMaker = field(default_factory=BaseAimsMaker)
+    criterion_name: str = "energy_per_atom"
+    epsilon: float = 0.001
+    convergence_field: str = field(default_factory=str)
+    convergence_steps: list = field(default_factory=list)
+
+    def __post_init__(self):
+        """Set the value of the last index."""
+        self.last_idx = len(self.convergence_steps)
+
+    def make(self, atoms):
```

Comment:
What is this make function for? Can you just rename `convergence_iteration` to `make` to make this consistent with other `Makers`.

Reply:
Can you see my comment at the top of this class. I think there is a better way to do this

Reply:
Is this make function actually used somewhere?

Reply:
>    Can you see my comment at the top of this class. I think there is a better way to do this

Sorry, there is no comment at the top of this class

>    Is this make function actually used somewhere?

Yes, you can only use make for this to work (see eg the test in `test_makers/test_convergence.py`). If you rename `convergence_iteration` to `make`, you'll not be able to replace jobs in the flow, because there will be no flow

Reply:
I'll make some changes to clarify what is going on here and then both of you can see if you agree with them

Reply:
I modified this in my last commit, I basically moved `convergence_iteration` into a separate job function independent of the `Maker`. I'll now test to see if `convergence_iteration` can be removed entirely with that logic included in make of the Convergence workflow

Reply:
I removed the `convergence_iteration` function entirely and moved it to main for the Convergence maker and made that `make` function a job and not a flow.

@utf and @ansobolev do you both agree with the current implementation.

Reply:
I kept the `detours` functionality over the `replace` one. Is this okay or should we modify this to use `replace?

Reply:
Y

> I removed the `convergence_iteration` function entirely and moved it to main for the Convergence maker and made that `make` function a job and not a flow.
> 
> @utf and @ansobolev do you both agree with the current implementation.

Yes, that's ok